services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -42,32 +44,37 @@
 
 from .config import (
     APP_CLIENT_REDIS_LOCK_KEY,
-    APP_GARBAGE_COLLECTOR_KEY,
     GUEST_USER_RC_LOCK_FORMAT,
     get_garbage_collector_interval,
 )
 from .registry import RedisResourceRegistry, get_registry
 
 logger = logging.getLogger(__name__)
+database_errors = (psycopg2.DatabaseError, asyncpg.exceptions.PostgresError)
 
 
-async def setup_garbage_collector_task(app: web.Application):
-    loop = asyncio.get_event_loop()
-    app[APP_GARBAGE_COLLECTOR_KEY] = loop.create_task(garbage_collector_task(app))
-    yield
-    task = app[APP_GARBAGE_COLLECTOR_KEY]
-    task.cancel()
-    await task
+def setup_garbage_collector(app: web.Application):
+    async def _setup_background_task(app: web.Application):
+        # on_startup
+        # create a background task to collect garbage periodically
+        loop = asyncio.get_event_loop()
+        cgp_task = loop.create_task(collect_garbage_periodically(app))
```

Comment:
Could you please rename `cgp_task` to `periodic_garbage_collect_task` or similar?

Reply:
yes. will do in next PR. I want to get thisone in so I can push py3.8 upgrade